components/message.rst

```diff
@@ -0,0 +1,284 @@
+.. index::
+   single: Message
+   single: Components; Message
+
+The Message Component
+=====================
+
+    The Message component helps application to send and receive messages
+    to/from other applications or via
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/message
+
+Alternatively, you can clone the `<https://github.com/symfony/message>`_ repository.
+
+.. include:: /components/require_autoload.rst.inc
+
+Concepts
+--------
+
+.. image:: /_images/components/message/overview.png
+
+**Sender**:
+   Responsible for serializing and sending the message to _something_. This
+   something can be a message broker or a third party API for example.
+
+**Receiver**:
+   Responsible for deserializing and forwarding the messages to handler(s). This
+   can be a message queue puller or an API endpoint for example.
+
+**Handler**:
+   Given a received message, contains the user business logic related to the
+   message. In practice, that is just a PHP callable.
+
+Bus
+---
+
+The bus is used to dispatch messages. MessageBus' behavior is in its ordered
+middleware stack. When using the message bus with Symfony's FrameworkBundle, the
+following middlewares are configured for you:
+
+#. ``LoggingMiddleware`` (logs the processing of your messages)
+#. ``SendMessageMiddleware`` (enables asynchronous processing)
+#. ``HandleMessageMiddleware`` (calls the registered handle)
+
+Example::
+
+    use App\Message\MyMessage;
+
+    $result = $this->get('message_bus')->handle(new MyMessage(/* ... */));
+
+Handlers
+--------
+
+Once dispatched to the bus, messages will be handled by a "message handler". A
+message handler is a PHP callable (i.e. a function or an instance of a class)
+that will do the required processing for your message. It _might_ return a
+result::
+
+    namespace App\MessageHandler;
+
+    use App\Message\MyMessage;
+
+    class MyMessageHandler
+    {
+       public function __invoke(MyMessage $message)
+       {
+           // Message processing...
+       }
+    }
+
+.. code-block:: xml
+
+    <service id="App\Handler\MyMessageHandler">
+       <tag name="message_handler" />
+    </service>
+
+.. note::
+
+    If the message cannot be guessed from the handler's type-hint, use the
+    ``handles`` attribute on the tag.
```

Comment:
I think best would be an example. Also mention that the guessing is done on the `_invoke` method argument